tree/ntuple/v7/src/RField.cxx

```diff
@@ -75,34 +113,14 @@ std::vector<std::string> TokenizeTypeList(std::string templateType) {
 }
 
 std::string GetNormalizedType(const std::string &typeName) {
-   std::string normalizedType(TClassEdit::CleanType(typeName.c_str(), /*mode=*/2));
-
-   // TODO(jblomer): use a type translation map
-   if (normalizedType == "Bool_t") normalizedType = "bool";
-   if (normalizedType == "Float_t") normalizedType = "float";
-   if (normalizedType == "Double_t") normalizedType = "double";
-   if (normalizedType == "Char_t") normalizedType = "char";
-   if (normalizedType == "int8_t") normalizedType = "std::int8_t";
-   if (normalizedType == "UChar_t") normalizedType = "std::uint8_t";
-   if (normalizedType == "unsigned char") normalizedType = "std::uint8_t";
-   if (normalizedType == "uint8_t") normalizedType = "std::uint8_t";
-   if (normalizedType == "Short_t") normalizedType = "std::int16_t";
-   if (normalizedType == "int16_t") normalizedType = "std::int16_t";
-   if (normalizedType == "UShort_t") normalizedType = "std::uint16_t";
-   if (normalizedType == "uint16_t") normalizedType = "std::uint16_t";
-   if (normalizedType == "Int_t") normalizedType = "std::int32_t";
-   if (normalizedType == "int") normalizedType = "std::int32_t";
-   if (normalizedType == "int32_t") normalizedType = "std::int32_t";
-   if (normalizedType == "unsigned") normalizedType = "std::uint32_t";
-   if (normalizedType == "unsigned int") normalizedType = "std::uint32_t";
-   if (normalizedType == "UInt_t") normalizedType = "std::uint32_t";
-   if (normalizedType == "uint32_t") normalizedType = "std::uint32_t";
-   if (normalizedType == "Long64_t") normalizedType = "std::int64_t";
-   if (normalizedType == "Long_t") normalizedType = "std::int64_t";
-   if (normalizedType == "int64_t") normalizedType = "std::int64_t";
-   if (normalizedType == "ULong64_t") normalizedType = "std::uint64_t";
-   if (normalizedType == "uint64_t") normalizedType = "std::uint64_t";
-   if (normalizedType == "string") normalizedType = "std::string";
+   std::string normalizedType(
+      TClassEdit::ResolveTypedef(TClassEdit::CleanType(typeName.c_str(),
+                                                       /*mode=*/2).c_str()));
+
+   auto translatedType = typeTranslationMap.find(normalizedType);
+   if (translatedType != typeTranslationMap.end())
+      normalizedType = translatedType->second;
+
    if (normalizedType.substr(0, 7) == "vector<") normalizedType = "std::" + normalizedType;
```

Comment:
Note that this is probably not enough (case of nested vector).  You could try to use `TClassEdit::InsertStd`.

Reply:
Thanks for the review, @pcanal!

I think we can leave it as-is, given that this function is executed once per created RField, i.e. for a `vector<vector<int>>`, it will run once for the outer type turning it into `std::vector<...>`, once for the inner vector turning it into `std::vector<int>`, and once for `int`, turning it into `std::int32_t`.